src/main/java/com/thealgorithms/scheduling/NonPreemptivePriorityScheduling.java

```diff
@@ -0,0 +1,126 @@
+package com.thealgorithms.scheduling;
+
+import java.util.PriorityQueue;
+
+/**
+ * This class implements the Non-Preemptive Priority Scheduling algorithm.
+ * Processes are executed in order of their priority. The process with the
+ * highest priority (lower priority number) is executed first,
+ * and once a process starts executing, it cannot be preempted.
+ */
+public final class NonPreemptivePriorityScheduling {
+
+    private NonPreemptivePriorityScheduling() {
+    }
+
+    /**
+     * Represents a process with an ID, burst time, priority, arrival time, and start time.
+     */
+    static class Process implements Comparable<Process> {
+        int id;
+        int arrivalTime;
+        int startTime;
+        int burstTime;
+        int priority;
+
+        /**
+         * Constructs a Process instance with the specified parameters.
+         *
+         * @param id          Unique identifier for the process
+         * @param arrivalTime Time when the process arrives in the system
+         * @param burstTime   Time required for the process execution
+         * @param priority    Priority of the process
+         */
+        Process(int id, int arrivalTime, int burstTime, int priority) {
+            this.id = id;
+            this.arrivalTime = arrivalTime;
+            this.startTime = -1;
+            this.burstTime = burstTime;
+            this.priority = priority;
+        }
+
+        /**
+         * Compare based on priority for scheduling. The process with the lowest
+         * priority is selected first.
+         * If two processes have the same priority, the one that arrives earlier is selected.
+         *
+         * @param other The other process to compare against
+         * @return A negative integer, zero, or a positive integer as this process
+         *         is less than, equal to, or greater than the specified process.
+         */
+        @Override
+        public int compareTo(Process other) {
+            if (this.priority == other.priority) {
+                return Integer.compare(this.arrivalTime, other.arrivalTime);
+            }
+            return Integer.compare(this.priority, other.priority);
+        }
+    }
+
+    /**
+     * Schedules processes based on their priority in a non-preemptive manner, considering their arrival times.
+     *
+     * @param processes Array of processes to be scheduled.
+     * @return Array of processes in the order they are executed.
+     */
+    public static Process[] scheduleProcesses(Process[] processes) {
+        PriorityQueue<Process> pq = new PriorityQueue<>();
+        int currentTime = 0;
+        int index = 0;
+        Process[] executionOrder = new Process[processes.length];
+
+        for (Process process : processes) {
+            pq.add(process);
+        }
+
+        while (!pq.isEmpty()) {
+            Process currentProcess = pq.poll();
+
+            if (currentTime < currentProcess.arrivalTime) {
```

Comment:
You just idle while waiting for the highest priority process to arrive, but should be executing other processes during that time https://www.educative.io/answers/what-is-a-non-preemptive-priority-scheduling-algorithm

Reply:
Added `waitingQueue` variable to address the above change.